upstream/upstream.go

```diff
@@ -66,3 +70,35 @@ const (
 
 	DefaultSemVerConstraints = ">= 0.0.0"
 )
+
+func selectHighestVersion(constraints string, expectedRange semver.Range, tags []string) (string, error) {
+	var candidateVersion semver.Version
+	candidateVersionString := "" // keep the version string separately as it may contain a leading `v`
+	for _, tag := range tags {
+		// Try to match semver and range
+		version, err := semver.Parse(strings.Trim(tag, "v"))
```

Comment:
We can use `util.TagStringToSemver` for that: https://github.com/kubernetes-sigs/release-utils/blob/441a2c4cd8b97d693318f3c207752df8335db66b/util/common.go#L380